tests/firedrake/regression/test_interp_dual.py

```diff
@@ -34,6 +34,13 @@ def f1(mesh, V1):
     return Function(V1).interpolate(expr)
 
 
+def test_interp_self(V1):
+    a = assemble(TestFunction(V1) * dx)
+    b = assemble(TestFunction(V1) * dx)
+    a.interpolate(a)
+    assert (a.dat.data_ro == b.dat.data_ro).all()
```

Comment:
Not sure if this should be a `np.allclose` test

Reply:
I think you're right as written. No flops should be performed so that should be an equality.

Reply:
I don't think it's actually no flops, it's interpolation which happens to be the identify except for roundoff.

Reply:
Changed to `np.allclose`, I'm pretty sure now that's correct